service/controller/chart/resource/status/resource.go

```diff
@@ -83,13 +84,13 @@ func equals(a, b v1alpha1.ChartStatus) bool {
 	if a.AppVersion != b.AppVersion {
 		return false
 	}
-	if a.Release.LastDeployed != b.Release.LastDeployed {
+	if !a.Release.LastDeployed.Equal(b.Release.LastDeployed) {
 		return false
 	}
 	if a.Reason != b.Reason {
 		return false
 	}
-	if a.Release.Revision != b.Release.Revision {
+	if to.Int(a.Release.Revision) != to.Int(b.Release.Revision) {
```

Comment:
This was all kinds of wrong. 🤦‍♂️ 

The revision became a pointer because it's optional. So we need to cast to an int for comparison.